R/derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,329 @@
+#' Derive Confirmation Flag
+#'
+#' Derive a flag which depends on subsequent observations of the dataset. For
+#' example, flagging events which need to be confirmed by a second event.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `by_vars` and `join_vars` parameter are
+#'   expected.
+#'
+#' @param by_vars By variables
+#'
+#'   The specified variables are used as by variables for joining the input
+#'   dataset with itself.
+#'
+#' @param order Order
+#'
+#'   The observations are ordered by the specified order.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param join_vars Variables to keep from joined dataset
+#'
+#'   The variables needed from the subsequent observations should be specified
+#'   for this parameter. The specified variables are added to the joined dataset
+#'   with suffix ".join". For example to flag all observations with `AVALC ==
+#'   "Y"` and `AVALC == "Y"` for at least one subsequent visit `join_vars =
+#'   vars(AVALC, AVISITN)` and `filter = AVALC == "Y" & AVALC.join == "Y" &
+#'   AVISITN < AVISITN.join` could be specified.
+#'
+#'   The `*.join` variables are not included in the output dataset.
+#'
+#' @param join_type Observations to keep after joining
```

Comment:
is this a new argument since the admiralonco version? i didn't notice it before or in the issue. i find it a bit confusing so think more is needed here in the description to explain why needed.

my understanding is when we did this for onco BOR we only ever needed to look for the records on or after first PR/CR here, but now for the long COVID example we'd actually want the option to look even before the AE of interest.

Reply:
Yes, this is a new argument to control which confirmation assessments are considered. I have added some more explanation. Is it clearer now?

Reply:
Minor suggested changes to your addition:

#'   For example for confirmed response or BOR in the oncology setting or
#'   confirmed deterioration in questionnaires the confirm**atory** assessment must be
#'   after the assessment to be flagged. Thus `join_type = "after"` could be
#'   used.
#'
**#'   Whereas, sometimes you might allow for confirmatory observations to 
#'   occur prior to the observation to be flagged. For example,
#'   to flag AEs occurring on or after seven days before a COVID AE.
#'   Thus `join_type = "all"` could be used.**